components/menu.js

```diff
@@ -0,0 +1,57 @@
+import Link from 'next/link';
+import { FaPlus, FaTimes } from 'react-icons/fa';
+import { useState } from 'react';
+
+export const Menu = ({ menu }) => {
+  const [toggle, setToggle] = useState(false);
+  const toggler = () => {
+    toggle ? setToggle(false) : setToggle(true);
+  };
+  return (
+    <>
+      <div className="menu">
+        {menu.map((menuItem) => (
+          <Link key={menuItem.id} href={menuItem.link}>
+            {menuItem.label}
+          </Link>
+        ))}
+      </div>
+      <div className="plus" onClick={toggler}>
+        {toggle ? <FaTimes size="5vh" /> : <FaPlus size="5vh" />}
+      </div>
+      <style jsx>
```

Comment:
an optimization tip: create 2 styles and collect all styles with variables and conditions into one and static ones to the other. by that way you will update only necessary styles the rest won't be updated